src/transform/classify.rs

```diff
@@ -0,0 +1,69 @@
+/// Transforms for classifying (tagging and categorizing) events.
+///
+/// Based on code in aw_research: https://github.com/ActivityWatch/aw-research/blob/master/aw_research/classify.py
+
+use regex::Regex;
+use serde_json;
+
+use std::collections::HashSet;
+use crate::models::Event;
+
+pub fn classify(mut events: Vec<Event>, classes: Vec<(String, Regex)>) -> Vec<Event> {
+    let mut classified_events = Vec::new();
+    for event in events.drain(..) {
+        classified_events.push(classify_one(event, &classes));
+    }
+    return classified_events;
+}
+
+fn classify_one(mut event: Event, classes: &Vec<(String, Regex)>) -> Event {
+    let mut tags: HashSet<String> = HashSet::new();
+    for (cls, re) in classes {
+        for val in event.data.values() {
+            // TODO: Recurse if value is object/array
+            if val.is_string() && re.is_match(val.as_str().unwrap()) {
+                tags.insert(cls.clone());
+                break;
+            }
+        }
+    }
+
+    // An event can have many tags
+    event.data.insert("$tags".into(), serde_json::json!(tags));
```

Comment:
What does the "$" mean?

Reply:
Just some prefix I chose for "computed property". I'm not sure if this is how we want to do it but it's how I've been doing it for a while and it's not that bad.

Reply:
Sounds  sane.
Maybe we should do the same with the fields we generate with "split_url_event" function which is already in aw-server?

Reply:
We should, I'll fix it in this batch of related PRs.